join.spec.md

```diff
@@ -100,204 +101,347 @@ flowchart TB
+  - Routers will often omit schema elements from the API schema that is used to validate client operations and that they present to clients via introspection ({join__Graph}, for example, will typically be omitted)
 
 <a name=def-endpoint>**Endpoints**</a> are running servers which can resolve GraphQL queries against a schema. In this version of the spec, endpoints must be URLs, typically http/https URLs.
 
 <a name=def-subgraph>**Subgraphs**</a> are GraphQL schemas which are composed to form a supergraph. Subgraph names and metadata are declared within the special {join__Graph} enum.
 
 This spec does not place any requirements on subgraph schemas. Generally, they may be of any shape. In particular, subgraph schemas do not need to be supergraphs themselves or to follow this spec in any way; neither is it an error for them to do so. Composers MAY place additional requirements on subgraph schemas to aid in composition; composers SHOULD document any such requirements.
 
-# Basic Requirements
+# Data Model
 
-Schemas using {@join} MUST be valid [core schema documents](/core/v0.1) with {@core} directives referencing this specification.
+TODO maybe this should be called something more like "Overview" or "Graph Routing in a Supergraph" or something?
 
-:::[example](photos.graphql#schema) -- {@core} directives for supergraphs
+*This section is non-normative.* It describes the motivation behind the directives defined by this specification.
 
-# Enums
+A supergraph schema describes a GraphQL schema that can be served by a router. The router does not contain logic to resolve any of the schema's fields; instead, the supergraph schema contain directives starting with {@join__} which tell the router which subgraph endpoints can resolve each field, as well as other information needed in order to construct subgraph queries.  The directives described in this specification are designed for a particular query planning algorithm, and so there are some restrictions on how they can be combined that originate from the requirements of this algorithm. (We hope that future versions of this specification can relax some of these restrictions.)
 
-##! join__Graph
+Each supergraph schema contains a list of the subgraphs. It represents this as an enum called [{join__Graph}](#join__Graph) with an enum value for each subgraph. Each enum value is annotated with a [{@join__graph}](#@join__graph) directive telling the router what endpoint can be used to reach the subgraph and a name for the subgraph that can be used in representations of query plans and diagnostic messages.
```

Comment:
I merged your commit.

I guess my point is saying that on the one hand, the name will be seen by actual humans, but on the other hand, it's not required to have a specific form or relate to anything else in particular (eg, the graphs themselves don't need to "know" their names). Any thoughts on making this more clear?

Uniqueness is mentioned normatively in the `@join__graph` section. 061aa3d tweaks the wording a bit

Reply:
I think naming it as a "human-readable identifier" tells me more clearly the point you're trying to get across and how it might be used.

..."that can be useful as a human-readable identifier in various use cases such as query plan visualization and server logs" (diagnostic messages is fine too)

Reply:
c3d8d12 on main